src/main/java/com/rabbitmq/perf/Consumer.java

```diff
@@ -103,22 +108,76 @@ public Consumer(Channel channel, String id,
             };
         }
 
-        this.state = new ConsumerState(rateLimit);
-        this.recoveryProcess = recoveryProcess;
+        this.state = new ConsumerState(parameters.getRateLimit());
+        this.recoveryProcess = parameters.getRecoveryProcess();
         this.recoveryProcess.init(this);
     }
 
     public void run() {
-        try {
-            q = new ConsumerImpl(channel);
-            for (String qName : queueNames) {
-                String tag = channel.basicConsume(qName, autoAck, q);
-                consumerTagBranchMap.put(tag, qName);
+        if (this.polling) {
```

Comment:
I'd recommend extracting this clause into a separate method, same for the `else` clause.